sklearn/utils/estimator_checks.py

```diff
@@ -2448,4 +2448,12 @@ def check_fit_idempotent(name, estimator_orig):
     for method in check_methods:
         if hasattr(estimator, method):
             new_result = getattr(estimator, method)(X_test)
-            assert_allclose_dense_sparse(result[method], new_result)
+            if np.issubdtype(new_result.dtype, np.floating):
+                tol = np.finfo(new_result.dtype).eps
+            else:
+                tol = np.finfo(np.float64).eps
```

Comment:
maybe add a comment `# integer dtype` because it's not clear at first sight.

I wonder if it would not be even clearer if you reformulate this way:
```python
if new_result.dtype == np.float32:
    tol = np.finfo(np.float32).eps
else:
    tol = np.finfo(np.float64).eps
```
wdyt ?

Reply:
I think using np.floating is much cleaner NumPy, it accounts for np.half, np.longdouble etc., even though these types may not be used in Scikit-Learn.

Reply:
fair enough :)